src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Extensions/PropertyExtensions.cs

```diff
@@ -7,23 +7,22 @@ namespace VirtoCommerce.XDigitalCatalog.Extensions
 {
     public static class PropertyExtensions
     {
-        public static IList<Property> ExpandByValues(this IEnumerable<Property> properties)
+        public static IList<Property> ExpandByValues(this IEnumerable<Property> properties, string cultureName)
         {
             var result = properties.SelectMany(property =>
             {
-                var propertyValues = property.Values.Select(v =>
-                {
-                    var clonedProperty = (Property)property.Clone();
-                    clonedProperty.Values = new List<PropertyValue> { v };
-                    return clonedProperty;
-                }).ToList();
+                var propertyValues = property.Values
+                    .Where(propertyValue => propertyValue.LanguageCode.EqualsInvariant(cultureName))
```

Comment:
Seems it is incorrect, you should return localized value only if it exists, otherwise value with default culture. What if you asked DE locale, we do not have it for some value - a property with empty value will be returned?